packages/dashboard/src/components/contextMenu/index.test.tsx

```diff
@@ -0,0 +1,378 @@
+import * as React from 'react';
+import { Provider } from 'react-redux';
+import { DndProvider } from 'react-dnd';
+import { TouchBackend } from 'react-dnd-touch-backend';
+
+import { act } from 'react-dom/test-utils';
+import { screen } from '@testing-library/dom';
+import { render } from '@testing-library/react';
+import UserEvent from '@testing-library/user-event';
+
+import ContextMenu, { ContextMenuProps } from './index';
+import { DefaultDashboardMessages } from '~/messages';
+import { DASHBOARD_CONTAINER_ID } from '../grid/getDashboardPosition';
+import { configureDashboardStore } from '~/store';
+
+const renderContextMenu = (args: ContextMenuProps, open?: boolean) => {
+  const ret: { container: HTMLElement | null } = {
+    container: null,
+  };
+
+  act(() => {
+    const { container } = render(
+      <Provider store={configureDashboardStore()}>
+        <DndProvider
+          backend={TouchBackend}
+          options={{
+            enableMouseEvents: true,
+            enableKeyboardEvents: true,
+          }}
+        >
+          <div id={DASHBOARD_CONTAINER_ID} style={{ height: '1000px', width: '1000px' }}>
+            <ContextMenu {...args} />
+          </div>
+        </DndProvider>
+      </Provider>
+    );
+    ret.container = container;
+  });
+
+  if (open) {
+    act(() => {
+      const ev = new MouseEvent('contextmenu', {
+        bubbles: true,
+        clientX: 10,
+        clientY: 10,
+      });
+      document.getElementById(DASHBOARD_CONTAINER_ID)?.dispatchEvent(ev);
+    });
+  }
+
+  return ret;
+};
+
+const clickOption = async (option: string) => {
+  const user = UserEvent.setup();
+  await act(async () => {
+    await user.click(screen.getByText(option));
+  });
+};
+
+describe('ContextMenu', () => {
+  it('can toggle', () => {
+    const copyWidgets = jest.fn();
```

Comment:
not: if we are not spying on these can we just use arrow functions and skip creating a const for these and inline then in the `args` like `copyWidgets: () => {}`.

I noticed this is copied for other tests so maybe worth just creating using `jest.fn()` for the one we want to spy on.

Reply:
Yes we can do that